org-code-javabuilder/lib/src/main/java/org/code/neighborhood/Painter.java

```diff
@@ -0,0 +1,144 @@
+package org.code.neighborhood;
+
+public class Painter {
+    private int xLocation;
+    private int yLocation;
+    private Direction direction;
+    private int remainingPaint;
+    private Grid grid;
+    private String id;
+
+    public Painter(int x, int y, String direction, int paint) {
+        this.xLocation = x;
+        this.yLocation = y;
+        this.direction = new Direction(direction);
+        this.remainingPaint = paint;
+        this.grid = Grid.getInstance();
+        this.id = "Painter-"+this.grid.registerPainter();
+
+        System.out.println("created painter");
+    }
+
+    // Turn the painter one compass direction left (i.e. North -> West)
+    public void turnLeft() {
+        this.direction.turnLeft();
+    }
+
+    // Move the painter one square forward in the direction the painter
+    // is facing
+    public void move() {
```

Comment:
I think moveForward() is a little bit clearer but this is probably part of the interface that can't be changed anymore?

Reply:
@hanwalden ^

Reply:
Yes, we shortened it because of the amount of typing, and it's tightly tied to the curriculum. Not something we can easily change.